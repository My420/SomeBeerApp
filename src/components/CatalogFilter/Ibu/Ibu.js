import React from 'react';
import styles from './Ibu.module.scss';
import validateNumberValue from '../../../utils/validateNumberValue';
import {
  IBU_LESS_PROP,
  IBU_MORE_PROP,
  IBU_BOTTOM_VALUE,
  IBU_TOP_VALUE
} from '../../../utils/constants';

class Ibu extends React.PureComponent {
  onIBUChange = evt => {
    const { onPropertyChange } = this.props;
    const value = validateNumberValue(
      evt.target.value,
      IBU_BOTTOM_VALUE,
      IBU_TOP_VALUE
    );
    const direction = evt.target.name;
    if (direction === 'ibuLess') {
      onPropertyChange(value, IBU_LESS_PROP);
    } else if (direction === 'ibuMore') {
      onPropertyChange(value, IBU_MORE_PROP);
    }
  };

  render() {
    // eslint-disable-next-line no-console
    console.log('render ========== IbuInput');

    const { lessValue, moreValue } = this.props;
    return (
      <fieldset className={styles.ibuField}>
        <legend className={styles.ibulegend}>
          International Bitterness Unit
        </legend>
        <label className={styles.ibuLabelMore} htmlFor="ibuMore">
          {'More'}
          <input
            id="ibuMore"
            name="ibuMore"
            className={styles.ibuInputMore}
            type="number"
            value={moreValue}
            onChange={this.onIBUChange}
          />
        </label>
        <label className={styles.ibuLabelLess} htmlFor="ibuLess">
          {'Less'}
          <input
            id="ibuLess"
            name="ibuLess"
            className={styles.ibuInputLess}
            type="number"
            value={lessValue}
            onChange={this.onIBUChange}
          />
        </label>
      </fieldset>
    );
  }
}

export default Ibu;
