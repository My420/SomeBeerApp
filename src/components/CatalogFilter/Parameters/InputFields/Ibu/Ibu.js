import React from 'react';
import PropTypes from 'prop-types';
import styles from './Ibu.module.scss';
import validateNumberValue from '../../../../../utils/validateNumberValue';
import {
  IBU_LESS_PROP,
  IBU_MORE_PROP,
  IBU_BOTTOM_VALUE,
  IBU_TOP_VALUE
} from '../../../../../utils/constants';

const Ibu = ({ lessValue, moreValue, onPropertyChange }) => {
  const onIBUChange = evt => {
    const value = validateNumberValue(
      evt.target.value,
      IBU_BOTTOM_VALUE,
      IBU_TOP_VALUE
    );
    const direction = evt.target.name;
    if (direction === 'ibuLess') {
      onPropertyChange(value, IBU_LESS_PROP);
    } else if (direction === 'ibuMore') {
      onPropertyChange(value, IBU_MORE_PROP);
    }
  };
  return (
    <fieldset className={styles.ibuField}>
      <legend className={styles.ibuLegend}>
        International Bitterness Unit
      </legend>
      <div className={styles.ibuInner}>
        <label className={styles.ibuLabelMore} htmlFor="ibuMore">
          <span className={styles.textMore}>more than</span>
          <input
            id="ibuMore"
            name="ibuMore"
            className={styles.ibuInputMore}
            type="number"
            value={moreValue}
            onChange={onIBUChange}
          />
        </label>
        <label className={styles.ibuLabelLess} htmlFor="ibuLess">
          <span className={styles.textLess}>less than</span>
          <input
            id="ibuLess"
            name="ibuLess"
            className={styles.ibuInputLess}
            type="number"
            value={lessValue}
            onChange={onIBUChange}
          />
        </label>
      </div>
    </fieldset>
  );
};

Ibu.propTypes = {
  lessValue: PropTypes.string.isRequired,
  moreValue: PropTypes.string.isRequired,
  onPropertyChange: PropTypes.func.isRequired
};

export default Ibu;
