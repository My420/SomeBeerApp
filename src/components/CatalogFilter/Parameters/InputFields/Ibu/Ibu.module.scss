@import '../../../../../sass/variables.scss';

.ibuField {
  padding-bottom: 10px;
  border: none;
  @include border-bottom;
}

.ibuInputMore,
.ibuInputLess {
  width: 60px;
  height: 25px;

  @include border;
  @include input;
}

.textMore,
.textLess {
  margin-right: 7px;
}

.ibuInner {
  display: flex;
  justify-content: space-around;
}

.ibuLegend {
  display: inline-block;
  text-align: center;
  font-weight: 700;
}

@media (max-width: 400px) {
  .ibuField {
    padding-bottom: 5px;
  }

  .ibuInputMore,
  .ibuInputLess {
    width: 40px;
    height: 20px;
    font-size: 14px;
  }
}
