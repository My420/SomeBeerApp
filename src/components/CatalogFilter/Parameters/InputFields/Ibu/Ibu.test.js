import React from 'react';
import { shallow } from 'enzyme';
import Ibu from './Ibu';

describe('test <Ibu /> component', () => {
  const mockFunc = jest.fn();
  const wrapper = shallow(
    <Ibu lessValue="20" moreValue="30" onPropertyChange={mockFunc} />
  );

  beforeEach(() => {
    mockFunc.mockReset();
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should render  inputs with attr name='ibuLess' with correct value", () => {
    expect(wrapper.find("input[name='ibuLess']").prop('value')).toBe('20');
  });

  test("should render  inputs with attr name='ibuMore' with correct value", () => {
    expect(wrapper.find("input[name='ibuMore']").prop('value')).toBe('30');
  });

  test("should call onPropertyChange prop when inputs with attr name='ibuLess' change", () => {
    wrapper
      .find("input[name='ibuLess']")
      .simulate('change', { target: { value: '40', name: 'ibuLess' } });
    expect(mockFunc.mock.calls.length).toBe(1);
    expect(mockFunc.mock.calls[0][0]).toBe('40');
  });

  test("should call onPropertyChange prop when inputs with attr name='ibuMore' change", () => {
    wrapper
      .find("input[name='ibuMore']")
      .simulate('change', { target: { value: '10', name: 'ibuMore' } });
    expect(mockFunc.mock.calls.length).toBe(1);
    expect(mockFunc.mock.calls[0][0]).toBe('10');
  });
});
