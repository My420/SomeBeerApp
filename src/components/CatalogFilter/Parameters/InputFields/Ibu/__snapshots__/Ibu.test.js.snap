// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <Ibu /> component should match snapshot 1`] = `
<fieldset
  className="ibuField"
>
  <legend
    className="ibuLegend"
  >
    International Bitterness Unit
  </legend>
  <div
    className="ibuInner"
  >
    <label
      className="ibuLabelMore"
      htmlFor="ibuMore"
    >
      <span
        className="textMore"
      >
        more than
      </span>
      <input
        className="ibuInputMore"
        id="ibuMore"
        name="ibuMore"
        onChange={[Function]}
        type="number"
        value="30"
      />
    </label>
    <label
      className="ibuLabelLess"
      htmlFor="ibuLess"
    >
      <span
        className="textLess"
      >
        less than
      </span>
      <input
        className="ibuInputLess"
        id="ibuLess"
        name="ibuLess"
        onChange={[Function]}
        type="number"
        value="20"
      />
    </label>
  </div>
</fieldset>
`;
